compiler-plugin/src/main/java/io/ballerina/stdlib/persist/compiler/PersistRecordValidator.java

```diff
@@ -308,14 +305,22 @@ private void validateEntityAnnotation(SyntaxNodeAnalysisContext ctx, TypeDefinit
                                 if (!name.equals(Constants.TABLE_NAME)) {
                                     validateEntityProperties(ctx, expressionNode.get(), tableFields);
                                 } else {
-                                    tableName = Utils.eliminateDoubleQuotes(expressionNode.get().toSourceCode().trim());
-                                    if (tableNames.contains(tableName)) {
-                                        reportDiagnosticInfo(ctx, mappingFieldNode.location(),
+                                    ExpressionNode exp = expressionNode.get();
+                                    if (exp instanceof BasicLiteralNode) {
+                                        tableName = Utils.eliminateDoubleQuotes(exp.toSourceCode().trim());
+                                        if (tableNames.contains(tableName)) {
+                                            reportDiagnosticInfo(ctx, mappingFieldNode.location(),
+                                                    DiagnosticsCodes.PERSIST_113.getCode(),
+                                                    DiagnosticsCodes.PERSIST_113.getMessage(),
+                                                    DiagnosticsCodes.PERSIST_113.getSeverity());
+                                        } else {
+                                            tableNames.add(tableName);
+                                        }
+                                    } else {
+                                        reportDiagnosticInfo(ctx, exp.location(),
```

Comment:
Are we continuing processing even if this validation is shown?

Reply:
Yes